### DISCUSSION TITLE
Moving jspsych-written online experiment to MTurk

### DISCUSSION THREAD

**Discussion by ontinglo** [jspsych/jsPsych#1989] [Q&A]
hi all, do you have any successful experiences in running a jspsych-written online experiment in Amazon MTurk? 

My usual practice is to run jspsych-written experiments on cognition.run so that I can share the link to social media. Now I want to collect much more data more efficiently therefore running the experiment on MTurk will be my solution.

The difficult part is that the relevant online information/tutorials are so limited...therefore, I am here to seek help.

What codes should I add in the jspsych experiment so that it could successfully interact with MTurk? Or do you have any online resources that could be shared with me so that I can learn it?

thank you so much for your help, i am struggling with this issue for a very long time and very desperate...

thanks

**Comment by jodeleeuw** [comment]
Hi @ontinglo,

It's hard to answer this question because it's not clear what part of the process you are having trouble with. It sounds like you need help figuring out how to set up a HIT on MTurk that will link to your cognition.run study. If so, you might look for guides that have been written about how to set up HITs that link to external surveys. There's nothing that's jsPsych specific at that point, so even if you find a guide about linking to a Qualtrics survey the same steps should generally apply.

FWIW, I find [Prolific](https://www.prolific.co/) to be a little more user friendly that MTurk, especially for new users. There's a guide in the docs about [intergrating with Prolific](https://www.jspsych.org/overview/prolific/).
